### PR TITLE
[WIP]Image prune test enablement

### PR DESCRIPTION
**Pull request by wzheng1**
This is for https://issues.redhat.com/browse/DEVEXP-434 and still working in progress.

### CODE REVIEW

File: vendor/github.com/onsi/gomega/go.mod [hunk 1]
```diff
@@ -13,3 +13,5 @@ require (
 	gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 // indirect
 	gopkg.in/yaml.v2 v2.2.1
 )
+
+go 1.13
```
Comment (adambkaplan) [inline vendor/github.com/onsi/gomega/go.mod:17]: @wzheng1 please revert this (not clear why it is here).

File: test/extended/images/hardprune.go [hunk 2]
```diff
@@ -23,323 +23,163 @@ import (
...
+	//imagesutil "github.com/openshift/origin/test/extended/images"
+
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
 var _ = g.Describe("[sig-imageregistry][Feature:ImagePrune][Serial][Suite:openshift/registry/serial][Local] Image hard prune", func() {
 	defer g.GinkgoRecover()
 	var oc = exutil.NewCLI("prune-images")
-	var originalAcceptSchema2 *bool
-	ctx := context.Background()
```
Comment (adambkaplan) [inline test/extended/images/hardprune.go:33]: For a first attempt at restoring these tests, I would not worry about hard pruning. Eventually we'll want this test back, but it isn't as pressing as ensuring the core pruning functions work.

@dmage would you agree?

Comment (dmage) [reply test/extended/images/hardprune.go:33]: Yes, I agree.

I'm fine to slightly update this test even if it stays Skipped if that helps to remove legacy code. We'll never have the docker socket anyway.

Comment (wzheng1) [reply test/extended/images/hardprune.go:33]: OK, I will not spend too much effort on hard pruning now. Should I revert my changes?

Comment (dmage) [reply test/extended/images/hardprune.go:33]: Let's focus on `oc adm prune images` and remove hard pruner changes from this PR. Keep them for a follow-up PR.

btw, you shouldn't have back merges in your PR, so you'll need to squash commits.

File: vendor/github.com/onsi/gomega/go.mod (hunk 1 above)
Comment (wzheng1) [reply vendor/github.com/onsi/gomega/go.mod]: Have reverted.

File: test/extended/images/hardprune.go (hunk 2 above)
Comment (wzheng1) [reply test/extended/images/hardprune.go]: OK, I have restored most part of hard prune, just leave some basic functions change which maybe used by prune.go.

**Pull request closed without merging**